packages/data-core/src/types/types.ts

```diff
@@ -0,0 +1,48 @@
+export interface Schema {
+  name: string // "Article"
+  displayAttribute: string
+  attributes: {
+    [field: string]: string | { type: string }
+  }
+  relationships?: {
+    [field: string]: {
+      type: "many" | "one"
+      schema: string
+    }
+  }
+}
+
+export interface QueryList {
+  fields?: string[]
+  page?: { size: number; number: number }
+  sort?: { [key: string]: "asc" | "desc" }
+  filter?: { [key: string]: { [filter: string]: string } }
+}
+
+export interface Record {
+  id: string
+  [key: string]: any // @todo strict typing
+}
+
+export interface Resource {
+  id: string
+  __schema: string
+  attributes: {
+    [key: string]: any // @todo
+  }
+  relationships?: {
+    [key: string]: any // @todo
+  }
+}
+
+export interface SourceConfig {
+  type: string
+  url: string
+}
+
+export interface SourceV0_0_0 {
+  version: "0.0.0"
```

Comment:
If we stick with incrementing numbers, we can use `>` / `<` and it will make that area of code simpler.